HW4_Mukhametshina/amino_acid_tools.py

```diff
@@ -0,0 +1,241 @@
+    'stop': ['uaa', 'uag', 'uga']
+}
+weight_amino = [71.08, 156.2, 114.1, 115.1, 103.1, 129.1, 128.1, 57.05, 137.1, 113.2, 113.2, 128.2, 131.2, 147.2, 97.12, 87.08,
+         101.1, 186.2, 163.2, 99.13, 168.05, 255.3,
+         71.08, 156.2, 114.1, 115.1, 103.1, 129.1, 128.1, 57.05, 137.1, 113.2, 113.2, 128.2, 131.2, 147.2, 97.12, 87.08,
+         101.1, 186.2, 163.2, 99.13, 168.05, 255.3]
+
+import random
+
+def long_amino_code(sequence):
+    """
+        Function translates a given sequence of one-letter amino acids
+    into a more understandable sequence of amino acids consisting of three letters
+
+        Parameters:
+            sequence (str): each letter refers to one-letter coded proteinogenic amino acids or "random"
+        Returns:
+            (str) translated in three-letter code
+    """
+    if sequence != 'random':
+        d_names = dict(zip(short_code, long_code))
+        recording = sequence.maketrans(d_names)
+        return sequence.translate(recording)
+    else:
+        len = int(input("введите желаемую длину: "))
+        bases = list(amino_acid)
+        amino_sequencqe = ''.join(random.choice(bases) for i in range(len))
+        d_names = dict(zip(short_code, long_code))
+        recording = amino_sequencqe.maketrans(d_names)
+        return "рандомная последовательнсть", amino_sequencqe, amino_sequencqe.translate(recording)
+
+def molecular_weight(sequence):
+    """
+    Function calculates molecular weight of the amino acid chain
+        Parameters:
+            sequence (str): each letter refers to one-letter coded proteinogenic amino acids or "random"
+        Returns:
+            weight (float) Molecular weight of tge given amino acid chain in Da
+        """
+    if sequence != 'random':
+        molecular_weights = dict(zip(short_code, weight_amino))
+        weight = sum(molecular_weights.get(aa, 0) for aa in sequence)
+        return weight
+    else:
+        len = int(input("введите желаемую длину: "))
+        bases = list(amino_acid)
+        amino_sequencqe = ''.join(random.choice(bases) for i in range(len))
+        molecular_weights = dict(zip(short_code, weight_amino))
+        weight = sum(molecular_weights.get(aa, 0) for aa in amino_sequencqe)
+        return "рандомная последовательнсть", amino_sequencqe, weight
+
+def amino_to_rna(amino_sequence):
+    """
+    Function translates an amino acid sequence into a possible RNA sequence
+        Parameters:
+                amino_sequence (str) or "random"
+        Returns:
+                (str) possible RNA sequence
+    """
+    if amino_sequence != 'random':
+        rna_sequence = ""
+
+        for aminoacid in amino_sequence:
+            if aminoacid in codon_table:
+                codons = codon_table[aminoacid]
+                # Selecting one random codon
+                codon = random.choice(codons)
+                rna_sequence += codon
+            else:
+                print("Unknown amino acid code: ", aminoacid)
```

Comment:
Наверное в таком случае было бы лучше упасть с ошибкой
```suggestion
                raise ValueError("Unknown amino acid code: ", aminoacid)
```
Только наверное проверка воода это опять же работа не этих функций, а лучше было бы сделать где то в самом начале в главной функции